crypto/src/main/scala/org/bitcoins/crypto/MuSig2Util.scala

```diff
@@ -92,21 +81,35 @@ object MuSig2Util {
         .reduce(_.add(_))
     }
 
-    lazy val secondKey: SchnorrPublicKey = keys.tail.head
+    // In truth this represents the first key different from the head key
+    lazy val secondKeyOpt: Option[SchnorrPublicKey] = {
+      keys.find(_ != keys.head)
```

Comment:
why not do `keys.tail.headOption`

Reply:
As I discovered while implementing test vectors, if the same pubkey is used in the 0,1 spots then you have to take the next one